feature/sessions/src/test/java/io/github/droidkaigi/confsched2023/sessions/SearchScreenTest.kt

```diff
@@ -0,0 +1,106 @@
+import io.github.droidkaigi.confsched2023.sessions.component.FilterLanguageChipTestTag
+import io.github.droidkaigi.confsched2023.sessions.component.FilterSessionTypeChipTestTag
+import io.github.droidkaigi.confsched2023.testing.HiltTestActivity
+import io.github.droidkaigi.confsched2023.testing.RobotTestRule
+import io.github.droidkaigi.confsched2023.testing.category.ScreenshotTests
+import io.github.droidkaigi.confsched2023.testing.robot.SearchScreenRobot
+import org.junit.Rule
+import org.junit.Test
+import org.junit.experimental.categories.Category
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.GraphicsMode
+import javax.inject.Inject
+
+@RunWith(AndroidJUnit4::class)
+@GraphicsMode(GraphicsMode.Mode.NATIVE)
+@HiltAndroidTest
+@Config(
+    qualifiers = RobolectricDeviceQualifiers.NexusOne,
+)
+class SearchScreenTest {
+
+    @get:Rule
+    @BindValue val robotTestRule: RobotTestRule = RobotTestRule<HiltTestActivity>(this)
+
+    @Inject
+    lateinit var searchScreenRobot: SearchScreenRobot
+
+    @Test
+    @Category(ScreenshotTests::class)
+    fun checkLaunchShot() {
+        searchScreenRobot {
+            setupSearchScreenContent()
+            checkScreenCapture()
+        }
+    }
+
+    @Test
+    @Category(ScreenshotTests::class)
+    fun checkFilterDayChipShot() {
+        searchScreenRobot {
+            checkFilterChipShot(FilterDayChipTestTag)
+        }
+    }
+
+    @Test
+    @Category(ScreenshotTests::class)
+    fun checkFilterCategoryChipShot() {
+        searchScreenRobot {
+            checkFilterChipShot(FilterCategoryChipTestTag)
+        }
+    }
+
+    @Test
+    @Category(ScreenshotTests::class)
+    fun checkFilterSessionTypeChipShot() {
+        searchScreenRobot {
+            checkFilterChipShot(FilterSessionTypeChipTestTag) {
+                scrollSearchFilterToLeft()
+            }
+        }
+    }
+
+    @Test
+    @Category(ScreenshotTests::class)
+    fun checkFilterLanguageChipShot() {
+        searchScreenRobot {
+            checkFilterChipShot(FilterLanguageChipTestTag) {
+                scrollSearchFilterToLeft()
+            }
+        }
+    }
+
+    private fun SearchScreenRobot.checkFilterChipShot(
+        filterChipTestTag: String,
+        scrollToLeft: (() -> Unit)? = null,
+    ) {
+        setupSearchScreenContent()
+        scrollToLeft?.invoke()
+
+        // select item
+        clickFilterChip(filterChipTestTag)
+        clickFirstDropdownMenuItem()
+        checkSearchScreenCapture()
+
+        // select other item
+        clickFilterChip(filterChipTestTag)
+        clickLastDropdownMenuItem()
+        checkSearchScreenCapture()
+
+        // remove all items
+        clickFilterChip(filterChipTestTag)
+        clickFirstDropdownMenuItem()
+        clickFilterChip(filterChipTestTag)
+        clickLastDropdownMenuItem()
+        checkSearchScreenCapture()
+    }
```

Comment:
When selecting a narrowing item in FilterChip, it is tested that the selected item name is reflected in FilterChip.